### PR TITLE
MC-40702: Order currency is used instead of bas currency in the shipping label for form

### LINKED ISSUE #31891: Order currency is used instead of bas currency in the shipping label for form
When the store has multiple currencies and order is placed in any other currency rather than the base currency then this issue happens. The shipping label form uses order currency amounts instead of base currency amounts.
I checked further how different shipping carriers (checked in DHL, UPS, and FedEx) handles submitted custom amount. All of them treats received a custom amount as a base amount. I have faced this issue on Magento version 2.3.4 and 2.4-develop.

Here is the case,

- My order was like below. Order is placed in USD and the base currency is AED.

 !3jMDQab.png|thumbnail! 

-  On the shipping label form "Customs Value" has a suffix of AED (base currency) but values filled in automatically are in USD

 !uyUV2gR.png|thumbnail! 

- Hance shipping label has invalid values. (Label should have 113AED as shown in the first screenshot) 

 !WHIbjUT.png|thumbnail! 

### Preconditions (*)
1. Magento 2.4-develop
1. Store should have more than one currency.
1. Need any online shipping method enabled like DHL, FedEx, UPS, or any other.

### Steps to reproduce (*)
1. Open the frontend and switch to any currency different currency than the base currency.

1. Place an order with any online shipping carrier. (I used DHL)
1. Got to the order page and open the new shipment page.
1. Check the checkbox "Create Shipping Label" and then "Submit Shipment..." - [screenshot](https://i.imgur.com/70CGPHU.png)

### Expected result (*)
1. All amounts shown on the shipping label should be in the base currency.

### Actual result (*)
1. All amounts shown on the shipping label are in the order currency.

## 
Please provide [Severity](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- < ] Severity: **S0** *- Affects critical data or functionality and leaves users without workaround.*
- [ ] Severity: **S1** *- Affects critical data or functionality and forces users to employ a workaround.*
- [x] Severity: **S2** *- Affects non-critical data or functionality and forces users to employ a workaround.*
- [ ] Severity: **S3** *- Affects non-critical data or functionality and does not force users to employ a workaround.*
- [ > Severity: **S4** *- Affects aesthetics, professional look and feel, “quality” or “usability”.*

### PR DESCRIPTION
**Pull request by engcom-Echo**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#31891

### Manual testing scenarios (*)

1. See magento/magento2#31891

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by yvaliya** [comment]
@engcom-Echo I think it won't resolve the actual issue mentioned #31891 because currency code isn't an issue. As I mentioned in the issue description, shipping carriers are treating the received amount as a base amount. So instead of changing the currency code, we need to change the amount displayed in the custom amount input field to the base amount of each item.

@zakdma @victorpetryk Please see the possible solution in #31941 instead.

File: dev/tests/integration/testsuite/Magento/Shipping/Block/Adminhtml/Order/AddToPackageTest.php [hunk 1]
```diff
@@ -42,30 +51,43 @@ protected function setUp(): void
...
 
-        return array_pop($items);
+    /**
+     * Verify currency code on custom value field
+     *
+     * @magentoDataFixture Magento/Shipping/_files/shipping_with_carrier_data_different_currency_code.php
+     */
+    public function testGetCurrencyCodeCustomValue ()
+    {
+        $expectedCurrencyCode = '<span class="customs-value-currency">
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Shipping/Block/Adminhtml/Order/AddToPackageTest.php]: This check is too unstable. Any change in formatting will break test
I Suggest extract value and check it
```
$template = '/<span class="customs-value-currency">\s*?(?<currency>[A-Za-z]+)\s*?<\/span>/';
$matches = [];
preg_match($template, $text, $matches);
$currency = $matches['currency'] ?? null;
assert...
```

File: dev/tests/integration/testsuite/Magento/Shipping/_files/shipping_with_carrier_data_different_currency_code.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+use Magento\TestFramework\Workaround\Override\Fixture\Resolver;
+use Magento\Framework\DB\Transaction;
+
+Resolver::getInstance()->requireDataFixture('Magento/Sales/_files/order_with_customer.php');
+
+$objectManager = Bootstrap::getObjectManager();
+/** @var Transaction $transaction */
+$transaction = $objectManager->get(Transaction::class);
+/** @var ProductRepositoryInterface $productRepository */
+$productRepository = $objectManager->create(ProductRepositoryInterface::class);
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Shipping/_files/shipping_with_carrier_data_different_currency_code.php]: please use get() for product repository

File: dev/tests/integration/testsuite/Magento/Shipping/_files/shipping_with_carrier_data_different_currency_code.php [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+$orderRepository->save($order);
+
+$shipmentItems = [];
+foreach ($order->getItems() as $orderItem) {
+    $shipmentItems[$orderItem->getId()] = $orderItem->getQtyOrdered();
+}
+$tracking = [
+    'carrier_code' => 'ups',
+    'title' => 'United Parcel Service',
+    'number' => '987654321'
```
Comment (zakdma) [inline dev/tests/integration/testsuite/Magento/Shipping/_files/shipping_with_carrier_data_different_currency_code.php]: add comma

**Review by zakdma — CHANGES_REQUESTED** [review]

**Comment by engcom-Echo** [comment]
@magento run all tests

**Comment by engcom-Echo** [comment]
@magento run Functional Tests CE

**Review by zakdma — APPROVED** [review]

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed
Tested by next scenario:
Additional currency added (UAH) and set as base currency
Online shipping method enabled ( DHL)
Order placed in USD
Shipping label created for order's shipping in admin
:x: **Before:**
Screenshot from 2021-02-04 12-48-14

:heavy_check_mark: **After:**
Screenshot from 2021-02-04 13-49-13

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31957`.

**Pull request merged**